lib/netext/grpcext/conn.go

```diff
@@ -204,7 +304,60 @@ func (c *Conn) NewStream(
 
 // Close closes the underhood connection.
 func (c *Conn) Close() error {
-	return c.raw.Close()
+	var err error
+	if c.addr != "" {
+		// Lock for mutating connections map
+		err = c.closeShared()
+		if errors.Is(err, errUnexpectedSharedConnection) {
+			err = c.raw.Close()
+		}
+	} else {
+		err = c.raw.Close()
+	}
+
+	return err
+}
+
+func (c *Conn) closeShared() error {
+	var (
+		addrMu      *sync.Mutex
+		ok          bool
+		addrMuEntry interface{}
+		err         error
+	)
+
+	// Lock for mutating connectionsAddrMu map
```

Comment:
I'm somewhat confused as to why we lock here, based on the fact that `connectionsAddrMu` and `connections` are `sync.Map` at the moment, which is designed to be safe for concurrent use without locking.

Could you expand the comment to reflect that information? 🙇🏻